d2/src/main/java/com/linkedin/d2/balancer/strategies/degrader/DegraderLoadBalancerStrategyV3.java

```diff
@@ -106,12 +108,39 @@ public DegraderLoadBalancerStrategyV3(DegraderLoadBalancerStrategyConfig config,
 
   }
 
+  @Override
+  public String getName()
+  {
+    return DEGRADER_STRATEGY_NAME;
+  }
+
+  private List<DegraderTrackerClient> castToDegraderTrackerClients(Map<URI, TrackerClient> trackerClients)
```

Comment:
Can we avoid this casting?

Reply:
Not really since the interface expects TrackerClient. Did you have something in mind?

Reply:
yeah -- see previous comments.

Reply:
Degrader strategy required DegraderTrackerClient so the cast is necessary. We can make an assumption and do a direct cast (potentially more dangerous) without for loop if you'd like.